### PR TITLE
Polish core module code

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

File: cupy/core/carray.pxi [hunk 1]
```diff
@@ -18,7 +18,7 @@ cdef class CArray(CPointer):
 
     def __init__(self, ndarray arr):
         cdef Py_ssize_t i
-        cdef int ndim = arr.ndim
+        cdef int ndim = arr._shape.size()
```
Comment (ir5) [inline cupy/core/carray.pxi]: Just curious, what is the benefit by rewriting this way? The two things look the same.

**Comment by ir5** [comment]
LGTM

**Comment by unnonouno** [comment]
test this please

**Comment by ir5** [comment]
I may be missing something, but what kinds of tests are necessary? This PR is about refactoring, and it does not add any new functions or something. I guessed it was okay simply if existing tests passed.

File: cupy/core/carray.pxi (hunk 1 above)
Comment (okuta) [inline cupy/core/carray.pxi]: sorry late reply. `ndim` is slow because it is Python property.

**Pull request merged**